### PR TITLE
[WIP] Linux Hardening Guide

### PR DESCRIPTION
**Pull request by tommytran732**
Signed-off-by: Tommy <contact@tommytran.io>

### CODE REVIEW

File: content/os/Linux Hardening Guide.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+- the Network (`share=network`) socket (internet access)
+- the PulseAudio socket (for both audio in and out), `device=all` (access to all devices including the camera)
+- `org.freedesktop.secrets` dbus (access to secrets stored on your keychain) for applications which do not need it
+
+If an application works natively with Wayland (and not running through the [XWayland](https://wayland.freedesktop.org/xserver.html) compatibility layer), consider revoking its access to the X11 (`socket=x11`) and [Inter-process communications (IPC)](https://en.wikipedia.org/wiki/Unix_domain_socket) socket (`share=ipc`) as well.
+
+Many flatpak apps come with broad filesystem permissions such as `filesystem=home` and `filesystem=host`. Some applications like implement the [Portals](https://docs.flatpak.org/en/latest/portal-api-reference.html) [API](https://en.wikipedia.org/wiki/API), which allows a file manager to pass files to the Flatpak application (e.g. VLC) without specific filesystem access privileges. Despite of this, many of them, including ones like VLC [still includes](https://github.com/flathub/org.videolan.VLC/blob/master/org.videolan.VLC.json) `filesystem=host` for no apparant reason.
```
Comment (WfKe9vLwSvv7rN) [inline content/os/Linux Hardening Guide.md:72]: `filesystem=host` is somewhat questionable, but `filesystem=home:ro` is necessary to work with playlist files. For VLC specifically, being a lot more than just a media player, it's quite possible that `filesystem=home:rw` might be necessary for some features. Users that don't need those functions in VLC should certainly limit the permissions.

Comment (tommytran732) [reply content/os/Linux Hardening Guide.md:72]: filesystem=home:ro is not great. The user should only grant access to the dirs where the playlists are.

**Comment by tommytran732** [comment]
Closing in favor of https://github.com/PrivSec-dev/privsec.dev/pull/36

File: content/os/Linux Hardening Guide.md (hunk 1 above)
Comment (Zanthed) [reply content/os/Linux Hardening Guide.md:72]: The point isn't why VLC needs an overly permissive permission, the point is that it does not need it and a safer, more secure, and more private option of using the filesystem picker exists.

Comment (WfKe9vLwSvv7rN) [reply content/os/Linux Hardening Guide.md:72]: > filesystem=home:ro is not great. The user should only grant access to the dirs where the playlists are.

I agree it is not great, but ultimately it's a tradeoff between usability and security. Most users will not think to grant VLC access to `~/Music` manually (or whatever other relevant directories); they will expect it to just work out of the box. It's much more reasonable to expect security-minded users to manually limit the permissions.

> The point isn't why VLC needs an overly permissive permission, the point is that it does not need it and a safer, more secure, and more private option of using the filesystem picker exists.

It is impossible to make playlist files work through Flatpak portals. That was my point in the first place. That is also why [Amberol (a Flatpak-only music player) does not support them](https://gitlab.gnome.org/World/amberol/-/issues/114).

**Pull request closed without merging**